doc/source/ray-core/miscellaneous.rst

```diff
@@ -216,3 +216,135 @@ To get information about the current available resource capacity of your cluster
+all the worker nodes connect to the head node, which adds a lot of entries to
+the ARP table. Ensure that the ARP cache size is large enough to handle this
+many nodes.
+Failure to do this will result in the head node hanging. When this happens,
+`dmesg` will show errors like `neighbor table overflow message`.
+
+In Ubuntu, the ARP cache size can be tuned in `/etc/sysctl.conf` by increasing
+the value of `net.ipv4.neigh.default.gc_thresh1` - `net.ipv4.neigh.default.gc_thresh3`. 
+For more details, please refer to the OS manual.
+
+Tuning Ray Settings
+~~~~~~~~~~~~~~~~~~~
+
+To run a large cluster, several parameters need to be tuned in Ray.
+
+Health check
+************
+
+GCS checks the health of the worker nodes by sending ping periodically.
+Sometimes, if the network is poor or the raylet is too busy to respond,
+the raylet might be marked dead incorrectly. 
+
+GCS uses a similar algorithm as k8s probes to check the healthiness of the
+Raylet: it sends the ping to the worker periodically and if it failed X times
+consecutively, the worker will be marked as dead. Once a worker is marked as
+dead, it'll be considered dead forever even it's not. The following OS
+environments can be used to tune this: 
+
+- `RAY_health_check_initial_delay_ms` The delay of the first health check, 5000ms by default.
+- `RAY_health_check_period_ms` The interval between two health check requests, 3000ms by default.
+- `RAY_health_check_timeout_ms` The timeout to consider a health check failed, 10000ms by default.
+- `RAY_health_check_failure_threshold` The consecutive failure threshold for the
+  worker node being considered dead, 5 by default. 
+
+
+Resource broadcasting
+*********************
+
+Another functionality GCS provided is to ensure each worker node has a view of
+available resources of each other in the Ray cluster. Each raylet is going to
+push its local available resource to GCS and GCS will broadcast it to all the
+raylet periodically. The time complexity is O(N^2). In a large Ray cluster, this
+is going to be an issue, since most of the time is spent on broadcasting the
+resources. There are several settings we can use to tune this: 
+
+- `RAY_resource_broadcast_batch_size` The maximum number of nodes in a single
+  request sent by GCS, by default 512.
+- `RAY_raylet_report_resources_period_milliseconds` The interval between two
+  resources report in raylet, 100ms by default.
+
+Be aware that this is a trade-off between scheduling performance and GCS loads.
+Decreasing the resource broadcasting frequency might make scheduling slower.
+
+### gRPC threads for GCS
+By default, only one gRPC thread is used for server and client polling from the
+completion queue. This might become the bottleneck if QPS is too high.
+
+- `RAY_gcs_server_rpc_server_thread_num` Control the number of threads in GCS
+  polling from the server completion queue, by default, 1. 
+- `RAY_gcs_server_rpc_client_thread_num` Control the number of threads in GCS
```

Comment:
Any reason we haven't set these higher by default?

Reply:
I think we previously got some release test broken. We are reenabling it in this PR (https://github.com/ray-project/ray/pull/30131/files#diff-66e8718c71d8ec5383a5cc83b0ade4d1aa8bad17d7ced811928bb666b4b63165)

But for users who are using ray 2.2, it won't be enabled. We can delete it later once it's merged and runs well.